### PR TITLE
 py-deap: was fixing setuptool version for no reason

### PR DESCRIPTION
**Pull request by cattabiani**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-deap/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-deap/package.py:18]:
```suggestion
    depends_on("py-setuptools", type="build")
    depends_on("py-setuptools@:57", type="build", when="@1.3.1")
```

**Review by matz-e — COMMENTED** [review]
I checked again, 1.3.1 uses 2to3, which requires an older setuptools. But the fix as proposed should work.

**Comment by heerener** [comment]
Maybe immedi

> I checked again, 1.3.1 uses 2to3, which requires an older setuptools. But the fix as proposed should work.

https://github.com/Thibauth/python-pushover/issues/42

57 should be good, no?

@cattabiani  I'd recommend filing this PR to upstream spack as well, or we risk losing the change in the next upgrade if they don't do it independently.

**Comment by cattabiani** [comment]
`57 should be good, no?` it is not good for the new spack and steps. Steps requires latest version

**Comment by cattabiani** [comment]
do I have pr powers in spack/spack? @heerener

**Comment by heerener** [comment]
> `57 should be good, no?` it is not good for the new spack and steps. Steps requires latest version

Makes sense - I mostly meant "good for 2to3 in deap 1.3.1" :-)

**Comment by heerener** [comment]
> do I have pr powers in spack/spack? @heerener

As far as I know, anyone can file a PR from a fork - it's how I've been doing all of mine :-)

**Review by matz-e — APPROVED** [review]

**Comment by cattabiani** [comment]
@matz-e what do I do? It looks like I cannot merge without prechecks/package-audits

**Comment by heerener** [comment]
> @matz-e what do I do? It looks like I cannot merge without prechecks/package-audits

It's probably just waiting for the GitLab pipeline - once that one passes it should merge.

**Pull request merged**
